libs/providers/flagd-web/src/lib/flagd-web-provider.ts

```diff
@@ -0,0 +1,240 @@
+    context: EvaluationContext
+  ): Promise<ResolutionDetails<boolean>> {
+    return this.evaluate(flagKey, defaultValue, context, this._promiseClient.resolveBoolean);
+  }
+
+  resolveStringEvaluation(
+    flagKey: string,
+    defaultValue: string,
+    context: EvaluationContext
+  ): Promise<ResolutionDetails<string>> {
+    return this.evaluate(flagKey, defaultValue, context, this._promiseClient.resolveString);
+  }
+
+  resolveNumberEvaluation(
+    flagKey: string,
+    defaultValue: number,
+    context: EvaluationContext
+  ): Promise<ResolutionDetails<number>> {
+    return this.evaluate(flagKey, defaultValue, context, this._promiseClient.resolveFloat);
+  }
+
+  resolveObjectEvaluation<U extends JsonValue>(
+    flagKey: string,
+    defaultValue: U,
+    context: EvaluationContext
+  ): Promise<ResolutionDetails<U>> {
+    return this.evaluate(flagKey, defaultValue, context, this._promiseClient.resolveObject);
+  }
+
+  private async evaluate<T extends FlagValue>(
+    flagKey: string,
+    defaultValue: T,
+    context: EvaluationContext,
+    resolver: (
+      request: PartialMessage<ResolveAnyRequest>,
+      options?: CallOptions | undefined
+    ) => Promise<ResolveAnyResponse>
+  ): Promise<ResolutionDetails<T>> {
+    try {
+      const result = await this._cache.get<T>(flagKey, context);
+      if (this._cacheEnabled && result) {
+        return result;
+      }
+      const res = await resolver({flagKey, context: Struct.fromJson(context as JsonValue)});
+      const details = {
+        // if this is a Struct (object resolution), parse it
+        value: (res.value && res.value instanceof Struct ? res.value.toJson() : res.value) as unknown as T,
+        reason: res.reason,
+        variant: res.variant,
+      };
+      if (this._cacheEnabled) {
+        this._cache.set(flagKey, context, details);
+      }
+      return details;
+    } catch (err) {
+      return {
+        reason: StandardResolutionReasons.ERROR,
+        errorCode: this.mapError(err),
+        value: defaultValue,
+      };
+    }
+  }
+
+  private updateCacheForFlagKey(message: EventStreamResponse) {
+    try {
+      if (message.data) {
+        const body = message.data.toJson() as unknown as ConfigurationChangeBody;
+        this._cache.flush(body.flagKey).then(() => {
+          this._logger?.debug(
+            `configuration change: ${body.type} ${body.flagKey} ${body.source}, associated cache items busted.`
+          );
+        });
+      }
+    } catch (err) {
+      this._logger?.warn(`Error updating cache.`);
+    }
+  }
+
+  private async retryConnect() {
+    this._delayMs = Math.min(this._delayMs * 3, this._maxDelay);
+    if (this._retry >= this._maxRetries) {
+      this._logger?.warn('Max retries reached. Giving up.');
+      return;
+    }
+    this._callbackClient.eventStream(
+      {},
+      (message) => {
+        this._logger?.debug(`event received: ${message.type}`);
+        switch (message.type) {
```

Comment:
Let me think about this. :thinking:

Reply:
There are some message types that occur (`keep-alive`) for instance, that are totally valid but also we don't react to. I think we don't need a `default` case for this reason, since we already `debug` log regardless, and getting such events isn't an error or unexpected - they just aren't relevant to the for beyond the internal implementation of the provider.